lib/perl/Genome/Disk/Detail/Allocation/Unarchiver.pm

```diff
@@ -126,33 +116,54 @@ sub unarchive {
                 $old_absolute_path, $allocation_object->absolute_path);
         }
 
-        unless ($allocation_object->_commit_unless_testing) {
-            confess "Could not commit!";
-        }
-    };
-    my $error = $@;
-
-    # finally blocks would be really sweet. Alas...
-    $allocation_lock->unlock() if $allocation_lock;
-    $shadow_allocation->delete();
-
-    if ($error) {
-        if ($target_path and -d $target_path and not $ENV{UR_DBI_NO_COMMIT}) {
-            Genome::Sys->remove_directory_tree($target_path);
-        }
-        confess "Could not unarchive, received error:\n$error";
-    } else {
         $allocation_object->add_note(
             header_text => 'unarchived',
             body_text => $self->reason,
         );
-        Genome::Timeline::Event::Allocation->unarchived($self->reason,
-            $allocation_object);
+        Genome::Timeline::Event::Allocation->unarchived($self->reason, $allocation_object);
         $allocation_object->status('active');
+
+        unless ($tx->commit() && $allocation_object->_commit_unless_testing) {
+            die 'failed to commit';
+        }
+    }
+    catch {
+        $tx->rollback();
```

Comment:
If `$tx->commit()` succeeded this could result in:
```
Cannot rollback a transaction that is committed. at /.../UR/Context/Transaction.pm line 150
```